core/authority_discovery/publisher/address_publisher.cpp

```diff
@@ -133,27 +153,25 @@ namespace kagome::authority_discovery {
     for (const auto &address : addresses) {
       PB_SPAN_ADD(record, addresses, address.getBytesAddress());
     }
-    TimestampScale time{std::chrono::nanoseconds{
-        std::chrono::system_clock::now().time_since_epoch()}
-                            .count()};
-    PB_SPAN_SET(*record.mutable_creation_time(),
-                timestamp,
-                scale::encode(time).value());
+    if (now) {
+      TimestampScale time{now->count()};
+      PB_SPAN_SET(*record.mutable_creation_time(),
+                  timestamp,
+                  scale::encode(time).value());
```

Comment:
Check encode result before getting value

Reply:
No, `encode` never fails

Reply:
The function declaration tells the opposite. It's outcome_result. There are other places in code where this result is checked. Today it never fails, tomorrow the implementation changes and it does, so you have to keep in mind this particular line of code each time scale::encode or smth it relies on is changed somehow? Otherwise change encode function header so it returns value, not result.